spec.bs

```diff
@@ -3013,47 +3040,57 @@ algorithms to achieve the outcomes described in the above explanatory content.
 </div>
 
 <div algorithm>
-  Create a new algorithm, called <dfn>Derive a permissions policy directly from a fenced frame
-  config instance</dfn>.
+  Create a new algorithm, called <dfn>Create a permissions policy for a fenced navigable</dfn>.
 
-  Given null or an [=element=] (|container|), this algorithm returns a new [=permissions policy=].
+  Given null or an [=element=] (|container|) and an [=origin=] (|origin|), this algorithm returns a
+  new [=Permissions Policy=].
 
-  1. Let |inherited policy| be a new [=ordered map=].
+  1. [=Assert=]: if not null, container is a [=fenced navigable container=].
 
-  1. Let |effective permissions| be an empty [=list=].
-  
   1. Let |fenced frame config| be |container|'s [=Node/node document=]'s [=navigable/active browsing
```

Comment:
It seems like there is a pre-existing bug in this algorithm where we unconditionally grab things off of `|container|` which may be null, right?

Reply:
I think allowing `|container|` to be set to null is incorrect.

> If container is a [fenced navigable container](https://github.com/WICG/fenced-frame/pull/140/files/207f86fd773cfcc629867cde249766c27c56d713#fenced-navigable-containers), then let policy be the result of running [create a permissions policy for a fenced navigable](https://github.com/WICG/fenced-frame/pull/140/files/207f86fd773cfcc629867cde249766c27c56d713#create-a-permissions-policy-for-a-fenced-navigable) given container and origin.

This line, which is the only place that `Create a permissions policy for a fenced navigable` is called, ensures that `|container|` will always be a fenced navigable container inside the algorithm. So instead I think we need to assert just that `|container|` is a fenced navigable container, and not allow `null` to be an option for the parameter.

Alternatively we can have the `|container|` parameter in the algorithm be a `fenced navigable container` rather than just an `element`, but I'm going to keep it just as an `element` to be consistent with [Create a Permissions Policy for a navigable](https://w3c.github.io/webappsec-permissions-policy/#create-for-navigable).